packages/ui-keyring/src/Keyring.ts

```diff
@@ -143,6 +154,22 @@ export class Keyring extends Base implements KeyringStruct {
       .map((address) => this.getAddress(address));
   }
 
+  getContract (address: string | Uint8Array): KeyringAddress {
+    return this.getAddress(address, 'contract');
+  }
+
+  getContracts (): Array<KeyringAddress> {
+    const available = this.contracts.subject.getValue();
+
+    return Object
+      .entries(available)
+      .filter(([, data]) => {
+        const { json: { meta: { contractMeta } } } = data;
+        return contractMeta && contractMeta.genesisHash === this.genesisHash;
```

Comment:
This filtering is probably better suited here - https://github.com/polkadot-js/ui/blob/master/packages/ui-keyring/src/observable/genericSubject.ts#L16

I believe it gets filtered here, but shows up in the subscriptions.

Reply:
I did most of the work to change it so genericSubject does the filtering but couldn't figure out the issue - was coming back with empty objects. If anyone wants to try solving - https://github.com/polkadot-js/ui/tree/ki-contracts-2